src/components/search-wizard/filters/Coordinates.jsx

```diff
@@ -0,0 +1,259 @@
+    setX(point?.x);
+    setY(point?.y);
+  }, []);
+
+  const getInputs = (type) => {
+    switch (type) {
+      case 'utm':
+        return (
+          <>
+            <RangeValidationInput
+              className="mt-2"
+              label="Easting"
+              max={700_000}
+              min={200_000}
+              onChange={setX}
+              required
+              type="number"
+            />
+            <RangeValidationInput
+              className="mt-2"
+              label="Northing"
+              max={4_700_000}
+              min={4_060_000}
+              onChange={setY}
+              required
+              type="number"
+            />
+          </>
+        );
+
+      case 'decimalDegrees':
+        return (
+          <>
+            <RangeValidationInput
+              className="mt-2"
+              label="Longitude"
+              min={-114.25}
+              max={-108.7}
+              onChange={setX}
+              required
+              type="number"
+            />
+            <RangeValidationInput
+              className="mt-2"
+              label="Latitude"
+              min={36.7}
+              max={42.18}
+              onChange={setY}
+              required
+              type="number"
+            />
+          </>
+        );
+
+      case 'degreesMinutesSeconds':
+        return <DegreesMinutesSeconds onChange={onDegreeMinutesChange} />;
+
+      default:
+        throw new Error(`Unknown coordinate type: ${type}`);
+    }
+  };
+
+  const [inputGeometry, setInputGeometry] = useState(null);
+  useEffect(() => {
+    if (point && point.x && point.y) {
+      setInputGeometry(
+        new Point({
+          x: point.x,
+          y: point.y,
+          spatialReference: {
+            wkid: coordinateType === 'utm' ? 26912 : 4326,
+          },
+        })
+      );
+    } else {
+      setInputGeometry(null);
+    }
+  }, [coordinateType, point]);
+
+  return (
+    <div>
+      <strong>Coordinate System</strong>
+      <Select
+        ariaLabel="Coordinate Type"
+        items={Object.keys(coordinateTypes).map((key) => ({
+          value: key,
+          label: coordinateTypes[key].label,
+        }))}
+        onValueChange={setCoordinateType}
+        value={coordinateType}
+      />
+      {getInputs(coordinateType)}
+      <Buffer
+        className="mt-2"
+        onChange={setBufferGeometry}
+        inputGeometry={inputGeometry}
+      />
+    </div>
+  );
+}
+
+Coordinates.propTypes = {
+  send: PropTypes.func.isRequired,
+};
+
+function DegreesMinutesSeconds({ onChange }) {
+  const [longitudeDegrees, setLongitudeDegrees] = useState(null);
+  const [longitudeMinutes, setLongitudeMinutes] = useState(null);
+  const [longitudeSeconds, setLongitudeSeconds] = useState(null);
+  const [latitudeDegrees, setLatitudeDegrees] = useState(null);
+  const [latitudeMinutes, setLatitudeMinutes] = useState(null);
+  const [latitudeSeconds, setLatitudeSeconds] = useState(null);
```

Comment:
**thought**: how many useStates is a sign to use a reducer?

Reply:
This was my solution to an endless render loop. Maybe a reducer is a better option? I'm not sure if the reducer state would trigger a render for any prop that changes...